core-util/Array.h

```diff
@@ -45,6 +46,12 @@ class Array {
     /** Create a new array
       */
     Array(): _head(NULL) {
+        _grow_capacity = 0;
+        _alloc_traits = {0};
+        _alignment = MBED_UTIL_POOL_ALLOC_DEFAULT_ALIGN;
+        _capacity = 0;
+        _elements = 0;
+        _element_size = 0;
```

Comment:
It didn't like me putting the defaults into the initialiser list , get -Wreorder warnings due to values being set in the init function. As this is all likely to change soon anyway as there are a number of optimisations we would like to do on this code, may as well leave it as is for now.